editor/editor_node.cpp

```diff
@@ -633,6 +633,8 @@ void EditorNode::_notification(int p_what) {
 		} break;
 
 		case NOTIFICATION_ENTER_TREE: {
+			get_tree()->set_disable_node_threading(true); // No node threading while running editor.
```

Comment:
I think this should be an (advanced) editor setting so node threading for the editor can be toggled on/off.

Given the editor uses a wide surface of the engine apis, this would help with identifying bugs and issues with the node threading implementation. As an example, the Android port of the editor has already allowed us to identify bugs and issues with the engine on Android platforms.

Reply:
What do you mean by GUI will not support threading?

> Keep in mind that the way threading works at scene level

From the proposal and the screenshot above, it looks like threading can be configured _within a scene_ granularly at the node level. 
So just based off the screenshot, it would seem possible to configure different sections/nodes within the `EditorNode`  to run on different threads. I agree there may not be a reason to do so, but nonetheless the capability is there.

Reply:
GUI will not support threading, because it does a lot of interactions at the tree level across nodes that can easily exit your thread workgroups and go somewhere else (like container resizing). The engine can catch this fine and throw an error, but ultimately if you have a large connected GUI you can't easily split it up in threads. It can be implemented at some point, but it would be a lot of work for very little benefit. Add to this that the GUI does very little amount of work in the process callbacks (which is what gets parallelized) so even if it could be parallelized the benefit would be very minimum.

Reply:
Thanks, that makes sense at the moment. 
Although it's worth pointing out that the use of Godot as a UI framework to develop applications is likely to increase over time, and most current UI frameworks also have the concept of splitting work across multiple threads, so that may be something worth considering in the future.

At the moment, yes that is not much of a concern.